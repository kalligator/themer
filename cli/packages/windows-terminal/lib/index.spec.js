const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('themer Windows Terminal theme generator', () => {
  const promisedFiles = Promise.all(render(colors));

  it('should generate well-formatted themes', async () => {
    const files = await promisedFiles;
    const hexMatcher = expect.stringMatching(/#[a-zA-Z0-9]{6}/);
    for (const file of files) {
      const theme = JSON.parse(file.contents.toString('utf8'));
      expect(theme).toMatchObject({
        background: hexMatcher,
        black: hexMatcher,
        blue: hexMatcher,
        brightBlack: hexMatcher,
        brightBlue: hexMatcher,
        brightCyan: hexMatcher,
        brightGreen: hexMatcher,
        brightPurple: hexMatcher,
        brightRed: hexMatcher,
        brightWhite: hexMatcher,
        brightYellow: hexMatcher,
        cursorColor: hexMatcher,
        cyan: hexMatcher,
        foreground: hexMatcher,
        green: hexMatcher,
        name: expect.stringContaining('Themer'),
        purple: hexMatcher,
        red: hexMatcher,
        selectionBackground: hexMatcher,
        white: hexMatcher,
        yellow: hexMatcher,
      });
    }
  });

  it('should provide installation instructions', async () => {
    const files = await promisedFiles;
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();

    const singleThemeInstructions = renderInstructions([files[1].name]);
    expect(singleThemeInstructions).toMatchSnapshot();
  });
});
