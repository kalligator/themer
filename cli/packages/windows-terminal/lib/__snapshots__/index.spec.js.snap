// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer Windows Terminal theme generator should provide installation instructions 1`] = `
"
1. Open the Windows Terminal settings (\`Ctrl\`-\`,\`)
2. Add the contents of 'themer-dark.json' and 'themer-light.json' to the \`schemes\` array in \`profile.json\`
3. Set the \`colorScheme\` property to the desired scheme name (\\"Themer Dark\\" or \\"Themer Light\\") in the profiles section of \`profile.json\`, e.g.:

    \\"profiles\\": {
      \\"defaults\\": {
        \\"colorScheme\\": \\"Themer Dark\\"
      }
    }
"
`;

exports[`themer Windows Terminal theme generator should provide installation instructions 2`] = `
"
1. Open the Windows Terminal settings (\`Ctrl\`-\`,\`)
2. Add the contents of 'themer-light.json' to the \`schemes\` array in \`profile.json\`
3. Set the \`colorScheme\` property to the desired scheme name (\\"Themer Light\\") in the profiles section of \`profile.json\`, e.g.:

    \\"profiles\\": {
      \\"defaults\\": {
        \\"colorScheme\\": \\"Themer Light\\"
      }
    }
"
`;
